inbm/dispatcher-agent/dispatcher/update_logger.py

```diff
@@ -187,6 +199,23 @@ def update_granular_with_sota_package_list_install(self) -> None:
             with open(GRANULAR_LOG_FILE, 'w') as f:
                 json.dump(data, f, indent=4)
 
+    def update_granular_with_log(self, log: dict) -> None:
+        """This function stores the log provided into the granular log file."""
+        logger.debug("")
+        try:
+            # Load current data in granular log file.
+            with open(GRANULAR_LOG_FILE, 'r') as f:
```

Comment:
is there any possibility we could be updating the log from multiple threads simultaneously? if so we would need to protect file access with appropriate locks

Reply:
We have a check to prevent duplicated OTA being triggered. To be safe, I will add the lock to prevent race condition.